docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,588 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
 
   import {symbol1 as alias, symbol2} from "filename";
 
-Paths
------
+.. _path-resolution:
 
-In the above, ``filename`` is always treated as a path with ``/`` as directory separator,
-and ``.`` as the current and ``..`` as the parent directory.  When ``.`` or ``..`` is followed by a character except ``/``,
-it is not considered as the current or the parent directory.
-All path names are treated as absolute paths unless they start with the current ``.`` or the parent directory ``..``.
+Path Resolution
+---------------
 
-To import a file ``filename`` from the same directory as the current file, use ``import "./filename" as symbolName;``.
-If you use ``import "filename" as symbolName;`` instead, a different file could be referenced
-(in a global "include directory").
+The paths used in imports are not filesystem paths.
+In the simplest cases they may end up being used directly to load files from disk but in general
+the final path can be substantially different due to abstractions necessary to ensure reproducible
+builds on different platforms.
 
-It depends on the compiler (see :ref:`import-compiler`) how to actually resolve the paths.
-In general, the directory hierarchy does not need to strictly map onto your local
-filesystem, and the path can also map to resources such as ipfs, http or git.
+.. index:: source unit ID, import path, filesystem path
+
+Virtual Filesystem
+~~~~~~~~~~~~~~~~~~
+
+The compiler maintains an internal database (virtual filesystem) where each compiled source unit is
+assigned a unique *source unit ID* which is an opaque and unstructured identifier.
+
+While a source unit ID in the virtual filesystem can be anything, it should be a valid path if the
+source is meant to be loaded from the underlying filesystem.
+When the requested ID is not present in the virtual filesystem, it is passed to the file loader.
+In case of the command-line compiler the file loader simply uses it as a path.
+The `JavaScript interface <https://github.com/ethereum/solc-js>`_ is a bit more flexible in that
+regard and allows the user to provide a callback to perform this operation.
+In this case the IDs can be arbitrary.
```

Comment:
The IDs can always be arbitrary, the cmdline interface is just unlikely to find appropriate files. Also standard-json allows arbitrary Ids doesn't it?